crypto/signing/ed25519/scalar.go

```diff
@@ -0,0 +1,147 @@
+package ed25519
+
+import (
+	"bytes"
+	"crypto/cipher"
+	"crypto/ed25519"
+
+	"github.com/ElrondNetwork/elrond-go/crypto"
+)
+
+type ed25519Scalar struct {
+	ed25519.PrivateKey
+}
+
+// Equal checks if the underlying private key inside the scalar objects contain the same bytes
+func (es *ed25519Scalar) Equal(s crypto.Scalar) (bool, error) {
+	privateKey, err := es.getPrivateKeyFromScalar(s)
+	if err != nil {
+		return false, err
+	}
+
+	return bytes.Equal(privateKey, es.PrivateKey), nil
+}
+
+// Set sets the underlying private key inside the scalar to the private key of the provided scalar
+func (es *ed25519Scalar) Set(s crypto.Scalar) error {
+	privateKey, err := es.getPrivateKeyFromScalar(s)
+	if err != nil {
+		return err
+	}
+
+	es.PrivateKey = privateKey
+
+	return nil
+}
+
+// Clone creates a new Scalar with same value as receiver
+func (es *ed25519Scalar) Clone() crypto.Scalar {
+	scalarBytes := make([]byte, len(es.PrivateKey))
+	copy(scalarBytes, es.PrivateKey)
+
+	return &ed25519Scalar{scalarBytes}
+}
+
+// GetUnderlyingObj returns the object the implementation wraps
+func (es *ed25519Scalar) GetUnderlyingObj() interface{} {
+	return es.PrivateKey
+}
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (es *ed25519Scalar) MarshalBinary() ([]byte, error) {
+	return es.PrivateKey, nil
+}
+
+// UnmarshalBinary decodes a scalar from its byte array representation and sets the receiver to this value
+func (es *ed25519Scalar) UnmarshalBinary(s []byte) error {
+	switch len(s) {
+	case ed25519.SeedSize:
+		es.PrivateKey = ed25519.NewKeyFromSeed(s)
+	case ed25519.PrivateKeySize:
+		es.PrivateKey = s
+	default:
+		return crypto.ErrInvalidPrivateKey
+	}
+
+	return nil
+}
+
+// SetInt64 is not needed for this use case, should be removed if possible
+//  - panics to prevent using it
+func (es *ed25519Scalar) SetInt64(v int64) {
+	panic("ed25519Scalar SetInt64 not implemented")	
```

Comment:
no panic in production code. should add log.error and do nothing.

Reply:
done